### PR TITLE
redact sensitive feature flag

### PR DESCRIPTION
**Pull request by liamaharon**
demo for @ZanCorDX 

Idea is to redact the error `Display` and `Debug` implementation where the error is defined, rather than creating new layers of wrapper we need to maintain and remember to use

### CODE REVIEW

File: .github/workflows/checks.yaml [hunk 1]
```diff
@@ -92,13 +99,13 @@ jobs:
           echo "RUSTC_WRAPPER=sccache" >> $GITHUB_ENV
 
       - name: Build the rbuilder
-        run: cargo build
+        run: cargo build --features="${{ matrix.features }}"
```
Comment (ferranbt) [inline .github/workflows/checks.yaml:102]: Are we doubling or CI times with this? Do we need to CI test that the logs are correct?

Comment (liamaharon) [reply .github/workflows/checks.yaml:102]: No, they run as seperate jobs in parallel. Yes ci should check all features.

**Review by ZanCorDX — APPROVED** [review]

**Review by metachris — APPROVED** [review]

File: crates/rbuilder/src/live_builder/block_output/relay_submit.rs [hunk 2]
```diff
@@ -474,7 +471,7 @@ async fn submit_bid_to_the_relay(
         }
         Err(SubmitBlockErr::SimError(err)) => {
             inc_failed_block_simulations();
-            error!(err = ?err, "Error block simulation fail, cancelling");
+            error!(err = ?err.clone(), "Error block simulation fail, cancelling");
```
Comment (ZanCorDX) [inline crates/rbuilder/src/live_builder/block_output/relay_submit.rs]: Leaking err!

**Review by ZanCorDX — CHANGES_REQUESTED** [review]
Did you forget to redact SubmitBlockErr?

**Comment by liamaharon** [comment]
> Did you forget to redact SubmitBlockErr?

Yes good catch, I missed that that also needs redacting from your previous PR. Fixing.

**Review by ZanCorDX — APPROVED** [review]

**Pull request merged**